playwright/src/main/java/com/microsoft/playwright/impl/BrowserTypeImpl.java

```diff
@@ -239,13 +193,17 @@ private BrowserContextImpl launchPersistentContextImpl(Path userDataDir, LaunchP
     if (options.acceptDownloads != null) {
       params.addProperty("acceptDownloads", options.acceptDownloads ? "accept" : "deny");
     }
+    params.add("selectorEngines", gson().toJsonTree(playwright.sharedSelectors.selectorEngines));
+    params.addProperty("testIdAttributeName", playwright.sharedSelectors.testIdAttributeName);
     JsonObject json = sendMessage("launchPersistentContext", params).getAsJsonObject();
+    BrowserImpl browser = connection.getExistingObject(json.getAsJsonObject("browser").get("guid").getAsString());
+    browser.connectToBrowserType(this, options.tracesDir);
     BrowserContextImpl context = connection.getExistingObject(json.getAsJsonObject("context").get("guid").getAsString());
     context.videosDir = options.recordVideoDir;
     if (options.baseURL != null) {
       context.setBaseUrl(options.baseURL);
```

Comment:
`baseURL` should now be available through `context.initializer.options.baseURL`.